enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,269 @@
+---
+title: scorecard-persistent-storage-option
+authors:
+  - "@jemccorm"
+reviewers:
+  - TBD
+  - "@estroz"
+  - "@jesusr"
+  - "@bparees"
+approvers:
+  - TBD
+creation-date: 2021-03-26
+last-updated: 2021-04-12
+status: implementable
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# scorecard-storage-option
+
+This enhancement provides scorecard users a means to persist reports or other output from their scorecard tests to a persistent location that can be collected
+by scorecard.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+None
+
+## Summary
+
+QE and other test writers sometimes require storage within their test environment (e.g. test container) to write reports and other test output material.  Today within scorecard this is not possible since a persistent volume is not available within the test container.  As the test completes, the container ends, leaving only the container log on the k8s cluster until the Pod is deleted.
+
+This enhancement would offer an optional mechanism whereby test writers have a persistent volume mounted within their test containers, provisioned, enabled, and cleaned up by scorecard.
```

Comment:
Is the idea here to allow the N+1th scorecard test to consume data from what the Nth test writes to storage, a process to read data from storage after all tests have executed, or both?

Reply:
scorecard tests would not be able to share storage with this proposal, they have storage they can use but its not visible to other tests, scorecard itself does have visibility into all the storage created for that test execution and acts as an aggregator of any test output that was written.

Reply:
In that case, will the storage objects created by scorecard from these specs have well-defined or discoverable names so external processes can look them up easily?

Reply:
handling post-processing of on-cluster storage is something no one really has asked for,  but scorecard could label the PVC's with a suite/test name to support that kind of use case.  This RFE instead has scorecard aggregating test output from the on-cluster storage onto the host storage where scorecard is running, the idea being test writers would want to see the output locally instead of feeding output into other on-cluster processing.

Reply:
I see. Perhaps a bullet point on this is worth adding to non-goals.

Reply:
I added a note on this in the non-goals section just now to better clarify.